### ISSUE TITLE
question about the model on huggingface

### ISSUE THREAD

**Issue by f-x-here** [VITA-Group/LLaGA#29]
Dear author,

I am trying to reproduce the results in your paper. However, due to the poor online support for Huggingface in mainland China, I attempted to download and cache the model you shared on Huggingface but found that files like pytorch-model.bin were missing. How can I obtain these files? I tried downloading the original vicuna-7b model directly and moving the config file you provided to the same directory to overwrite the original config for training, but I encountered a CUDA memory out error when running the training code. Since flash_atten does not support multi-GPU setups with RTX4090s, I can only use one RTX4090. Can a single RTX4090 with 24GB support the training? I saw other workers' responses in the issue, suggesting that using a single 24GB GPU is feasible.

I will attach screenshots of the error and my environment. Looking forward to your reply. Thank you

![Image](https://github.com/user-attachments/assets/59595764-865e-4f3c-8dc0-c1fad1d99519)

I tried this train code : CUDA_VISIBLE_DEVICES=0 ./scripts/train.sh vicuna nc cora 16 sbert

but encountered

![Image](https://github.com/user-attachments/assets/4235f22f-9a9a-4ad5-8fd9-f58576499a68)

I find in issue that said the comsumption is not very huge
![Image](https://github.com/user-attachments/assets/e311fef1-6f18-4218-b2c8-1985604afc6b)

**Comment by f-x-here** [comment]
and here is my enviroment
# packages in environment at /dfs/users_dic/cyy/anaconda3/envs/llaga:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
_sysroot_linux-64_curr_repodata_hack 3                   haa98f57_10  
accelerate                0.32.1                   pypi_0    pypi
aiohappyeyeballs          2.6.1                    pypi_0    pypi
aiohttp                   3.11.18                  pypi_0    pypi
aiosignal                 1.3.2                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
async-timeout             5.0.1                    pypi_0    pypi
attrs                     25.3.0                   pypi_0    pypi
binutils_impl_linux-64    2.40                 h5293946_0  
binutils_linux-64         2.40.0               hc2dff05_2  
blas                      1.0                         mkl  
brotli-python             1.0.9           py310h6a678d5_9  
bzip2                     1.0.8                h5eee18b_6  
ca-certificates           2025.2.25            h06a4308_0  
certifi                   2025.1.31          pyhd8ed1ab_0    conda-forge
charset-normalizer        3.3.2              pyhd3eb1b0_0  
click                     8.1.8                    pypi_0    pypi
cuda-cccl                 11.8.89                       0    nvidia/label/cuda-11.8.0
cuda-command-line-tools   11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-compiler             11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-cudart               11.8.89                       0    nvidia
cuda-cudart-dev           11.8.89                       0    nvidia/label/cuda-11.8.0
cuda-cuobjdump            11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-cupti                11.8.87                       0    nvidia
cuda-cuxxfilt             11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-documentation        11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-driver-dev           11.8.89                       0    nvidia/label/cuda-11.8.0
cuda-gdb                  11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-libraries            11.8.0                        0    nvidia
cuda-libraries-dev        11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-memcheck             11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-nsight               11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-nsight-compute       11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-nvcc                 11.8.89                       0    nvidia
cuda-nvdisasm             11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-nvml-dev             11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-nvprof               11.8.87                       0    nvidia/label/cuda-11.8.0
cuda-nvprune              11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-nvrtc                11.8.89                       0    nvidia
cuda-nvrtc-dev            11.8.89                       0    nvidia/label/cuda-11.8.0
cuda-nvtx                 11.8.86                       0    nvidia
cuda-nvvp                 11.8.87                       0    nvidia/label/cuda-11.8.0
cuda-profiler-api         11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-runtime              11.8.0                        0    nvidia
cuda-sanitizer-api        11.8.86                       0    nvidia/label/cuda-11.8.0
cuda-toolkit              11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-tools                11.8.0                        0    nvidia/label/cuda-11.8.0
cuda-version              12.8                          3    nvidia
cuda-visual-tools         11.8.0                        0    nvidia/label/cuda-11.8.0
datasets                  3.5.0                    pypi_0    pypi
deepspeed                 0.11.1                   pypi_0    pypi
dill                      0.3.8                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
einops                    0.7.0                    pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
filelock                  3.17.0          py310h06a4308_0  
flash-attn                1.0.9                    pypi_0    pypi
freetype                  2.13.3               h4a9f257_0  
frozenlist                1.6.0                    pypi_0    pypi
fsspec                    2024.12.0                pypi_0    pypi
gcc_impl_linux-64         11.2.0               h1234567_1  
gcc_linux-64              11.2.0               h5c386dc_2  
gds-tools                 1.4.0.31                      0    nvidia/label/cuda-11.8.0
gitdb                     4.0.12                   pypi_0    pypi
gitpython                 3.1.44                   pypi_0    pypi
gmp                       6.3.0                h6a678d5_0  
gmpy2                     2.2.1           py310h5eee18b_0  
gnutls                    3.6.15               he1e5248_0  
gxx_impl_linux-64         11.2.0               h1234567_1  
gxx_linux-64              11.2.0               hc2dff05_2  
hjson                     3.1.0                    pypi_0    pypi
huggingface-hub           0.30.2                   pypi_0    pypi
idna                      3.7             py310h06a4308_0  
intel-openmp              2023.1.0         hdb19cb5_46306  
jinja2                    3.1.6           py310h06a4308_0  
joblib                    1.4.2                    pypi_0    pypi
jpeg                      9e                   h5eee18b_3  
kernel-headers_linux-64   3.10.0              h57e8cba_10  
lame                      3.100                h7b6447c_0  
lcms2                     2.16                 h92b89f2_1  
ld_impl_linux-64          2.40                 h12ee557_0  
lerc                      4.0.0                h6a678d5_0  
libcublas                 11.11.3.6                     0    nvidia
libcublas-dev             11.11.3.6                     0    nvidia/label/cuda-11.8.0
libcufft                  10.9.0.58                     0    nvidia
libcufft-dev              10.9.0.58                     0    nvidia/label/cuda-11.8.0
libcufile                 1.13.1.3                      0    nvidia
libcufile-dev             1.4.0.31                      0    nvidia/label/cuda-11.8.0
libcurand                 10.3.9.90                     0    nvidia
libcurand-dev             10.3.0.86                     0    nvidia/label/cuda-11.8.0
libcusolver               11.4.1.48                     0    nvidia
libcusolver-dev           11.4.1.48                     0    nvidia/label/cuda-11.8.0
libcusparse               11.7.5.86                     0    nvidia
libcusparse-dev           11.7.5.86                     0    nvidia/label/cuda-11.8.0
libdeflate                1.22                 h5eee18b_0  
libffi                    3.4.4                h6a678d5_1  
libgcc                    14.2.0               h767d61c_2    conda-forge
libgcc-devel_linux-64     11.2.0               h1234567_1  
libgcc-ng                 14.2.0               h69a702a_2    conda-forge
libgomp                   14.2.0               h767d61c_2    conda-forge
libiconv                  1.16                 h5eee18b_3  
libidn2                   2.3.4                h5eee18b_0  
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
libnpp                    11.8.0.86                     0    nvidia
libnpp-dev                11.8.0.86                     0    nvidia/label/cuda-11.8.0
libnvjpeg                 11.9.0.86                     0    nvidia
libnvjpeg-dev             11.9.0.86                     0    nvidia/label/cuda-11.8.0
libpng                    1.6.39               h5eee18b_0  
libstdcxx-devel_linux-64  11.2.0               h1234567_1  
libstdcxx-ng              11.2.0               h1234567_1  
libtasn1                  4.19.0               h5eee18b_0  
libtiff                   4.7.0                hde9077f_0  
libunistring              0.9.10               h27cfd23_0  
libuuid                   1.41.5               h5eee18b_0  
libwebp-base              1.3.2                h5eee18b_1  
llvm-openmp               14.0.6               h9e868ea_0  
lz4-c                     1.9.4                h6a678d5_1  
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2           py310h5eee18b_0  
mdurl                     0.1.2                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46344  
mkl-service               2.4.0           py310h5eee18b_2  
mkl_fft                   1.3.11          py310h5eee18b_0  
mkl_random                1.2.8           py310h1128e8f_0  
mpc                       1.3.1                h5eee18b_0  
mpfr                      4.2.1                h5eee18b_0  
mpmath                    1.3.0           py310h06a4308_0  
multidict                 6.4.3                    pypi_0    pypi
multiprocess              0.70.16                  pypi_0    pypi
ncurses                   6.4                  h6a678d5_0  
nettle                    3.7.3                hbbd107a_1  
networkx                  3.4.2           py310h06a4308_0  
ninja                     1.11.1.4                 pypi_0    pypi
nltk                      3.9.1                    pypi_0    pypi
nsight-compute            2022.3.0.22                   0    nvidia/label/cuda-11.8.0
numpy                     1.24.1                   pypi_0    pypi
nvidia-ml-py              12.570.86          pyhd8ed1ab_0    conda-forge
nvitop                    1.4.2              pyh707e725_0    conda-forge
openh264                  2.1.1                h4ff587b_0  
openjpeg                  2.5.2                h0d4d230_1  
openssl                   3.5.0                h7b32b05_0    conda-forge
packaging                 23.2                     pypi_0    pypi
pandas                    2.2.0                    pypi_0    pypi
pathtools                 0.1.2                    pypi_0    pypi
peft                      0.5.0                    pypi_0    pypi
pillow                    11.1.0          py310hac6e08b_1  
pip                       25.0            py310h06a4308_0  
propcache                 0.3.1                    pypi_0    pypi
protobuf                  4.25.6                   pypi_0    pypi
psutil                    7.0.0           py310ha75aee5_0    conda-forge
py-cpuinfo                9.0.0                    pypi_0    pypi
pyarrow                   19.0.1                   pypi_0    pypi
pydantic                  1.10.21                  pypi_0    pypi
pyg-lib                   0.4.0+pt21cu118          pypi_0    pypi
pygments                  2.19.1                   pypi_0    pypi
pyparsing                 3.2.3                    pypi_0    pypi
pysocks                   1.7.1           py310h06a4308_0  
python                    3.10.16              he870216_1  
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.10                    2_cp310    conda-forge
pytorch                   2.1.0           py3.10_cuda11.8_cudnn8.7.0_0    pytorch
pytorch-cuda              11.8                 h7e8668a_6    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2025.2                   pypi_0    pypi
pyyaml                    6.0.2           py310h5eee18b_0  
readline                  8.2                  h5eee18b_0  
regex                     2024.11.6                pypi_0    pypi
requests                  2.32.3          py310h06a4308_1  
rich                      14.0.0                   pypi_0    pypi
safetensors               0.5.3                    pypi_0    pypi
scikit-learn              1.3.1                    pypi_0    pypi
scipy                     1.12.0                   pypi_0    pypi
sentence-transformers     2.2.2                    pypi_0    pypi
sentencepiece             0.2.0                    pypi_0    pypi
sentry-sdk                2.26.1                   pypi_0    pypi
setproctitle              1.3.5                    pypi_0    pypi
setuptools                75.8.0          py310h06a4308_0  
shortuuid                 1.0.11                   pypi_0    pypi
six                       1.17.0                   pypi_0    pypi
smmap                     5.0.2                    pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0  
sympy                     1.13.3          py310h06a4308_1  
sysroot_linux-64          2.17                h57e8cba_10  
tbb                       2021.8.0             hdb19cb5_0  
threadpoolctl             3.6.0                    pypi_0    pypi
tk                        8.6.14               h39e8969_0  
tokenizers                0.13.3                   pypi_0    pypi
torch-cluster             1.6.3+pt21cu118          pypi_0    pypi
torch-geometric           2.6.1                    pypi_0    pypi
torch-scatter             2.1.2+pt21cu118          pypi_0    pypi
torch-sparse              0.6.18+pt21cu118          pypi_0    pypi
torch-spline-conv         1.2.2+pt21cu118          pypi_0    pypi
torchaudio                2.1.0               py310_cu118    pytorch
torchtriton               2.1.0                     py310    pytorch
torchvision               0.16.0              py310_cu118    pytorch
tqdm                      4.67.1                   pypi_0    pypi
transformers              4.31.0                   pypi_0    pypi
trl                       0.5.0                    pypi_0    pypi
typing_extensions         4.12.2          py310h06a4308_0  
tzdata                    2025.2                   pypi_0    pypi
urllib3                   2.3.0           py310h06a4308_0  
wandb                     0.15.12                  pypi_0    pypi
wheel                     0.41.3                   pypi_0    pypi
xxhash                    3.5.0                    pypi_0    pypi
xz                        5.6.4                h5eee18b_1  
yaml                      0.2.5                h7b6447c_0  
yarl                      1.20.0                   pypi_0    pypi
zlib                      1.2.13               h5eee18b_1  
zstd                      1.5.6                hc292b87_0
